src/sentry/eventstream/snuba.py

```diff
@@ -98,6 +99,44 @@ def insert(
         if unexpected_tags:
             logger.error("%r received unexpected tags: %r", self, unexpected_tags)
 
+        # HACK: We are putting all this extra information that is required by the
+        # post process forwarder into the headers so we can skip parsing entire json
+        # messages. The post process forwarder is currently bound to a single core.
+        # Once we are able to parallelize the JSON parsing and other transformation
+        # steps being done there we may want to remove this hack.
+        def encode_bool(value: Optional[bool]) -> str:
+            if value is None:
+                value = False
+            return str(int(value))
+
+        # WARNING: We must remove all None headers. There is a bug in confluent-kafka-python
+        # (used by both Sentry and Snuba) that incorrectly decrements the reference count of
+        # Python's None on any attempt to read header values containing null values, leading
+        # None to eventually get deallocated and crash the interpreter.
+        def strip_none_values(value: Mapping[str, Optional[str]]) -> Mapping[str, str]:
+            return {key: value for key, value in value.items() if value is not None}
+
+        send_new_headers = options.get("eventstream:kafka-headers")
+
+        if send_new_headers is True:
+            headers = strip_none_values(
+                {
```

Comment:
This is a optimization nit, but only two fields could be None in this list. It seems unnecessary to iterate and rebuild the entire dictionary just for those two fields. I would build the initial headers, and then add those fields afterwards if they exist.